src/historywork/DownloadApplyTxsWork.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "historywork/DownloadApplyTxsWork.h"
+#include "catchup/ApplyCheckpointWork.h"
+#include "history/FileTransferInfo.h"
+#include "history/HistoryManager.h"
+#include "historywork/GetAndUnzipRemoteFileWork.h"
+#include "ledger/LedgerManager.h"
+#include "work/WorkSequence.h"
+
+namespace stellar
+{
+
+DownloadApplyTxsWork::DownloadApplyTxsWork(
+    Application& app, TmpDir const& downloadDir, LedgerRange range,
+    LedgerHeaderHistoryEntry& lastApplied)
+    : Work(app, "download-apply-ledgers", RETRY_NEVER)
+    , mRange(range)
+    , mDownloadDir(downloadDir)
+    , mLastApplied(lastApplied)
+    , mCurrCheckpoint(
+          app.getHistoryManager().checkpointContainingLedger(range.first()))
+{
+}
+
+DownloadApplyTxsWork::~DownloadApplyTxsWork()
+{
+}
+
+BasicWork::State
+DownloadApplyTxsWork::doWork()
+{
+    // This will properly abort the remaining children
+    if (anyChildRaiseFailure())
+    {
+        for (auto const& w : mRunning)
+        {
+            w->wakeUp();
```

Comment:
not sure I understand why you have to manage children directly: this should be handled by `Work` already?

Reply:
Yes, this is redundant (and was removed in the last commit)